### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by mikezange**

### CODE REVIEW

**Review by lushc — CHANGES_REQUESTED** [review]
Nice and tidy refactor 👍  just requesting a change to fix `ExampleSha256`

File: hasher_test.go [hunk 1]
```diff
@@ -102,79 +115,79 @@ func TestHmac(t *testing.T) {
...
+			t.Errorf("Expected %v, got %v", encoded, result)
+		}
+	})
 }
 
 func Example_hex() {
-	hash, err := Sha512([]byte("hello"), Hex)
-	fmt.Println(hash, err)
+	h := Sha512([]byte("hello")).Hex()
+	fmt.Println(h)
```
Comment (jordanfinners) [inline hasher_test.go:139]: How comes these are in a test file?

Comment (mikezange) [reply hasher_test.go:139]: https://blog.golang.org/examples

File: hasher_test.go [hunk 2]
```diff
@@ -102,79 +115,79 @@ func TestHmac(t *testing.T) {
...
 
 func ExampleSha1() {
-	hash, err := Sha1([]byte("hello"), Hex)
-	fmt.Println(hash, err)
+	h := Sha1([]byte("hello")).Hex()
+	fmt.Println(h)
 }
 
 func ExampleSha256() {
-	hash, err := Sha512([]byte("hello"), Hex)
```
[the remark this answers is not in the document]
Comment (mikezange) [reply hasher_test.go:150]: Fixed

**Review by lushc — APPROVED** [review]

File: hasher_test.go (hunk 1 above)
Comment (jordanfinners) [reply hasher_test.go]: Ooo not seen that before, good to know

**Review by lushc — APPROVED** [review]

**Pull request merged**